operator/src/main/java/oracle/kubernetes/operator/DomainStatusUpdater.java

```diff
@@ -564,47 +563,143 @@ private void setStatusConditions(DomainStatus status) {
         }
       }
 
-      private void addTooManyReplicasFailures(DomainStatus status) {
-        getConfiguredClusters().stream()
-              .map(TooManyReplicasCheck::new)
-              .filter(TooManyReplicasCheck::isFailure)
-              .forEach(check -> status.addCondition(check.createFailureCondition()));
+      private boolean haveServerData() {
+        return this.serverState != null;
       }
 
-      private List<WlsClusterConfig> getConfiguredClusters() {
-        return Optional.ofNullable(config).map(WlsDomainConfig::getConfiguredClusters).orElse(Collections.emptyList());
+      class Conditions {
+
+        private final ClusterCheck[] clusterChecks = createClusterChecks();
+        private final List<DomainCondition> conditions = new ArrayList<>();
+
+        void apply(DomainStatus status) {
+          status.removeConditionsMatching(c -> c.hasType(Failed) && ReplicasTooHigh.name().equals(c.getReason()));
+          conditions.forEach(status::addCondition);
```

Comment:
Shall we remove all Failed conditions in "status" if we have Completed condition with "True" status in "conditions"?

Reply:
Yes, that makes sense.

Reply:
We create the Completed condition, set to true or false, on every make right and status update. The case you describe is not relevant.

Reply:
What is the expected behavior for the case that I described? What is the expected status of the Completed condition after a fully running domain is updated with a weblogic credential setting that points to a non-existing secret?

Reply:
In such a case, we will have a failure, and that will result in completed/false.

Reply:
Thanks. 

Is the answer to my original question still true at this point? If so, do we need to address it in this PR?

Reply:
I think I didn't give you a good answer, originally. The statuses are recomputed every time we do the status update. If there are failures, we do not reach completed/true, so there is no need to remove them if we do reach that state - as it means that they have already been cleared.